Panic on reading artifactory_permission_target resource

**Describe the bug**
A terraform plan with no changes expected (see snippet) panics refreshing state of the permission_target resource

```terraform
resource "artifactory_local_repository" "localrepo1" {
  key             = "localrepo1"
  package_type    = "generic"
  repo_layout_ref = "simple-default"
}

resource "artifactory_user" "publish_user" {
  name                       = "publish_user"
  email                      = "example@example.com"
  groups                     = ["readers"]
  admin                      = false
  disable_ui_access          = true
  internal_password_disabled = true
}

resource "artifactory_permission_target" "publish_user" {
  name = "publish_user"
  repo {
    includes_pattern = ["**"]
    repositories = [
      "localrepo1"
    ]
    actions {
      users {
        name        = artifactory_user.publish_user.name
        permissions = ["annotate", "read", "write", "delete"]
      }
    }
  }
}
```

```bash
$ terraform plan -target artifactory_local_repository.localrepo1
No changes. Your infrastructure matches the configuration.

$ terraform plan -target artifactory_user.publish_user
No changes. Your infrastructure matches the configuration.

$ terraform plan -target artifactory_permission_target.publish_user
│ Error: Request cancelled
│ 
│   with artifactory_permission_target.publish_user,
│   on main.tf line 16, in resource "artifactory_permission_target" "publish_user":
│   16: resource "artifactory_permission_target" "publish_user" {
│ 
│ The plugin.(*GRPCProvider).ReadResource request was cancelled.

Stack trace from the terraform-provider-artifactory_v2.3.4 plugin:

panic: interface conversion: interface {} is []string, not string

goroutine 45 [running]:
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.HashString(...)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.0/helper/schema/set.go:17
github.com/jfrog/terraform-provider-artifactory/pkg/artifactory.hashPrincipal(0xf95540, 0xc000434ab0, 0x1140cd0)
	github.com/jfrog/terraform-provider-artifactory/pkg/artifactory/resource_artifactory_permission_target.go:134 +0x12d
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Set).hash(0xc000691a80, 0xf95540, 0xc000434ab0, 0x0, 0x1)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.0/helper/schema/set.go:251 +0x3d
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Set).add(0xc000691a80, 0xf95540, 0xc000434ab0, 0x0, 0xf95540, 0xc000434ab0)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.0/helper/schema/set.go:231 +0x85
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Set).Add(...)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.0/helper/schema/set.go:75
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*MapFieldReader).readSet.func1(0xc00003c990, 0x26, 0xc00044a998, 0x4, 0x1)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.0/helper/schema/field_reader_map.go:154 +0x30b
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.BasicMapReader.Range(0xc000434360, 0xc0002b6960, 0x16)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.0/helper/schema/field_reader_map.go:196 +0xc7
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*MapFieldReader).readSet(0xc0004814e8, 0xc0006defa0, 0x5, 0x5, 0xc0005e7680, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.0/helper/schema/field_reader_map.go:131 +0x4b2
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*MapFieldReader).ReadField(0xc0004814e8, 0xc0006defa0, 0x5, 0x5, 0x0, 0x0, 0x0, 0x0, 0x0, 0x128, ...)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.0/helper/schema/field_reader_map.go:31 +0x505
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.readObjectField(0x128a240, 0xc0004814e8, 0xc00057bc80, 0x4, 0x4, 0xc0000f0630, 0x0, 0x0, 0x0, 0x0, ...)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.0/helper/schema/field_reader.go:236 +0x21a
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*MapFieldReader).ReadField(0xc0004814e8, 0xc00057bc80, 0x4, 0x4, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.0/helper/schema/field_reader_map.go:33 +0x3f4
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.readListField(0x128a240, 0xc0004814e8, 0xc000434990, 0x3, 0x3, 0xc0005e8500, 0x0, 0x0, 0x0, 0x0, ...)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.0/helper/schema/field_reader.go:203 +0x36f
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*MapFieldReader).ReadField(0xc0004814e8, 0xc000434990, 0x3, 0x3, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.0/helper/schema/field_reader_map.go:27 +0x192
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.readObjectField(0x128a240, 0xc0004814e8, 0xc000691940, 0x2, 0x2, 0xc0000f0600, 0x0, 0x0, 0x0, 0x0, ...)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.0/helper/schema/field_reader.go:236 +0x21a
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*MapFieldReader).ReadField(0xc0004814e8, 0xc000691940, 0x2, 0x2, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.0/helper/schema/field_reader_map.go:33 +0x3f4
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.readListField(0x128a240, 0xc0004814e8, 0xc0004303d0, 0x1, 0x1, 0xc0005e7b80, 0x0, 0x0, 0x0, 0x0, ...)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.0/helper/schema/field_reader.go:203 +0x36f
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*MapFieldReader).ReadField(0xc0004814e8, 0xc0004303d0, 0x1, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0, 0xc0003d15e8, ...)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.0/helper/schema/field_reader_map.go:27 +0x192
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*MultiLevelFieldReader).ReadFieldMerge(0xc000691920, 0xc0004303d0, 0x1, 0x1, 0x10e8e5f, 0x3, 0x0, 0x0, 0x0, 0x0, ...)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.0/helper/schema/field_reader_multi.go:45 +0x1ff
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*ResourceData).get(0xc0000b5e00, 0xc0004303d0, 0x1, 0x1, 0x452108, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.0/helper/schema/resource_data.go:550 +0x34a
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*ResourceData).State(0xc0000b5e00, 0xc0000b5e00)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.0/helper/schema/resource_data.go:351 +0x32f
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).ShimInstanceStateFromValue(0xc000298b60, 0x12a0a10, 0xc000130ff8, 0xf95540, 0xc000434300, 0x12a0a10, 0xc000130ff8, 0xf95540)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.0/helper/schema/resource.go:260 +0x1f0
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadResource(0xc000128798, 0x129fee8, 0xc00057b900, 0xc00057b940, 0xc00057b900, 0x40b965, 0x1051380)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.0/helper/schema/grpc_provider.go:550 +0x279
github.com/hashicorp/terraform-plugin-go/tfprotov5/server.(*server).ReadResource(0xc0002af100, 0x129ff90, 0xc00057b900, 0xc0002b6240, 0xc0002af100, 0xc00037a930, 0xc0002ebba0)
	github.com/hashicorp/terraform-plugin-go@v0.3.0/tfprotov5/server/server.go:298 +0x105
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadResource_Handler(0x1099b60, 0xc0002af100, 0x129ff90, 0xc00037a930, 0xc0002b61e0, 0x0, 0x129ff90, 0xc00037a930, 0xc000414000, 0x171)
	github.com/hashicorp/terraform-plugin-go@v0.3.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:344 +0x214
google.golang.org/grpc.(*Server).processUnaryRPC(0xc00021ce00, 0x12a9b18, 0xc000135c80, 0xc000158700, 0xc0005f64e0, 0x18938f0, 0x0, 0x0, 0x0)
	google.golang.org/grpc@v1.32.0/server.go:1194 +0x52b
google.golang.org/grpc.(*Server).handleStream(0xc00021ce00, 0x12a9b18, 0xc000135c80, 0xc000158700, 0x0)
	google.golang.org/grpc@v1.32.0/server.go:1517 +0xd0c
google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc000114520, 0xc00021ce00, 0x12a9b18, 0xc000135c80, 0xc000158700)
	google.golang.org/grpc@v1.32.0/server.go:859 +0xab
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.32.0/server.go:857 +0x1fd

Error: The terraform-provider-artifactory_v2.3.4 plugin crashed!

```

**Requirements for and issue**
- [x] A fully functioning terraform snippet that can be copy&pasted (no outside files or ENV vars unless that's part of the issue)
- [x] Your version of artifactory = 7.5.7
- [x] Your version of terraform  = 1.0.3


**Expected behavior**
Successful plan showing no-changes.

**Additional context**
Plan is successful on provider v2.2.15

--- a/GNUmakefile
+++ b/GNUmakefile
@@ -1,5 +1,6 @@
 TEST?=./...
-PKG_NAME=pkg/artifactoVERSION := $(shell git tag --sort=-creatordate | head -1 | sed  -n 's/v\([0-9]*\).\([0-9]*\).\([0-9]*\)/\1.\2.\3/p')
+PKG_NAME=pkg/artifactory
+VERSION := $(shell git tag --sort=-creatordate | head -1 | sed  -n 's/v\([0-9]*\).\([0-9]*\).\([0-9]*\)/\1.\2.\3/p')
 NEXT_VERSION := $(shell echo ${VERSION}| awk -F '.' '{print $$1 "." $$2 "." $$3 +1 }' )
 
 default: build

--- a/pkg/artifactory/resource_artifactory_permission_target.go
+++ b/pkg/artifactory/resource_artifactory_permission_target.go
@@ -3,6 +3,7 @@ package artifactory
 import (
 	"fmt"
 	"net/http"
+	"strings"
 
 	"github.com/go-resty/resty/v2"
 	"github.com/jfrog/jfrog-client-go/artifactory/services"
@@ -131,7 +132,10 @@ func resourceArtifactoryPermissionTarget() *schema.Resource {
 
 func hashPrincipal(o interface{}) int {
 	p := o.(map[string]interface{})
-	return schema.HashString(p["name"].(string)) + 31*schema.HashString(schema.HashString(castToStringArr(p["permissions"].(*schema.Set).List())))
+	part1 := schema.HashString(p["name"].(string)) + 31
+	permissions := castToStringArr(p["permissions"].(*schema.Set).List())
+	part3 := schema.HashString(strings.Join(permissions,""))
+	return part1 * part3
 }
 
 func unpackPermissionTarget(s *schema.ResourceData) *services.PermissionTargetParams {

--- a/pkg/artifactory/resource_artifactory_permission_target_test.go
+++ b/pkg/artifactory/resource_artifactory_permission_target_test.go
@@ -100,7 +100,65 @@ const permissionFull = `
      //depends_on = [artifactory_local_repository.{{ .repo_name }}]
 	}
 `
-
+func TestGitHubIssue126(test *testing.T) {
+	_, permFqrn, permName := mkNames("test-perm", "artifactory_permission_target")
+	_, _, repoName := mkNames("test-perm-repo", "artifactory_local_repository")
+	_, _, username := mkNames("artifactory_user", "artifactory_user")
+	testConfig := `
+		resource "artifactory_local_repository" "{{ .repo_name }}" {
+		  key             = "{{ .repo_name }}"
+		  package_type    = "generic"
+		  repo_layout_ref = "simple-default"
+		}
+		
+		resource "artifactory_user" "{{ .username }}" {
+		  name                       = "{{ .username }}"
+		  email                      = "example@example.com"
+		  groups                     = ["readers"]
+		  admin                      = false
+		  disable_ui_access          = true
+		  internal_password_disabled = true
+		  password 					 = "Password1"
+		}
+		
+		resource "artifactory_permission_target" "{{ .perm_name }}" {
+		  name = "{{ .perm_name }}"
+		  repo {
+			includes_pattern = ["**"]
+			repositories = [
+			  "{{ .repo_name }}"
+			]
+			actions {
+			  users {
+				name        = artifactory_user.{{ .username }}.name
+				permissions = ["annotate", "read", "write", "delete"]
+			  }
+			}
+		  }
+		}`
+	variables := map[string]string{
+		"perm_name" : permName,
+		"username" : username,
+		"repo_name" : repoName,
+	}
+	foo := executeTemplate(permFqrn, testConfig, variables)
+	resource.Test(test, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(test) },
+		CheckDestroy: testPermissionTargetCheckDestroy(permFqrn),
+		Providers:    testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: foo,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(permFqrn, "name", permName),
+					resource.TestCheckResourceAttr(permFqrn, "repo.0.actions.0.users.#", "1"),
+					resource.TestCheckResourceAttr(permFqrn, "repo.0.actions.0.users.0.permissions.#", "4"),
+					resource.TestCheckResourceAttr(permFqrn, "repo.0.repositories.#", "1"),
+				),
+			},
+		},
+	})
+}
 func TestAccPermissionTarget_full(test *testing.T) {
 	_, permFqrn, permName := mkNames("test-perm", "artifactory_permission_target")
 	//_, _, repoName := mkNames("test-perm-repo", "artifactory_local_repository")

--- a/sample.tf
+++ b/sample.tf
@@ -13,7 +13,7 @@ variable "supported_repo_types" {
     "alpine",
     "bower",
     // xray refuses to cargo. They also require a mandatory field we can't currently support
-    "cargo",
+//    "cargo",
     "chef",
     "cocoapods",
     "composer",

